FutureNorthants/northants-website
# Add image paragraph slice

## Updates
- Add image slice to embedded paragraph and accordion embedded paragraph
- Extract some common methods to traits to help reduce duplication
- Added styles for better display in the editor and the view in the CMS

## To Test
- Clone branch and run `./drush.wrapper @docker config:import` 
- Edit a Service page and add an image to the body
- Edit a Service page and add an image to an accordion slice
- Check the below query returns the expected data for the Image Paragraph and that the image url sizes are of the correct ratio (4by3 or 16by9)
- Check the view page in the CMS and see that the image is visible with the caption below

```graphql
{
  route(path: "/pth/to/test/page") {
    __typename
    ... on DrupalNodeRoute {
      node {
        __typename
        ... on ServicePageNode {
          __typename
          title
          id
          serviceBody: body {
            value
            embeds {
              id
              paragraph {
                __typename
                ... on ImageParagraph {
                  __typename
                  caption
                  imageLarge
                  imageSmall
                  imageAltText
                  ratio
                }
                ... on AccordionParagraph {
                  __typename
                  sections {
                    id
                    title
                    summary
                    body {
                      value
                      embeds {
                        id
                        paragraph {
                          __typename
                          ... on ImageParagraph {
                            __typename
                            imageLarge
                            imageSmall
                            imageAltText
                            caption
                            ratio
                          }
                        }
                        __typename
                      }
                      __typename
                    }
                    __typename
                  }
                }
              }
              __typename
            }
            __typename
          }
        }
      }
      __typename
    }
  }
}
```
